gobblin-service/src/main/java/org/apache/gobblin/service/monitoring/KafkaJobStatusMonitor.java

```diff
@@ -90,6 +100,10 @@
   @Getter
   private final StateStore<org.apache.gobblin.configuration.State> stateStore;
   private final ScheduledExecutorService scheduledExecutorService;
+  private static final Config RETRYER_FALLBACK_CONFIG = ConfigFactory.parseMap(ImmutableMap.of(
+      RETRY_TIME_OUT_MS, TimeUnit.HOURS.toMillis(24L), // after a day, presume non-transient and give up
```

Comment:
we could consider indefinite retries, although I wanted to set a finite boundary to avoid "infinite looping".  if the error we're presuming transient takes >24 hours to resolve it probably merits human intervention.  at that point an operator is there to restart the service.  realistically, the probability of a truly transient error continuing >24 hours seems low.

if actually a problem with the message itself and/or incompatibility against the DB schema, that may require manual advance of the offset to skip the message or to adjust systems.  concern we might misdiagnose something like that as a transient error made me want to prevent an infinite loop.

still, while this is finite, it may always be overridden by config:
```
  public static final String KAFKA_JOB_STATUS_MONITOR_RETRY_TIME_OUT_MINUTES =
      "gobblin.kafka.jobStatusMonitor.retry.timeOut.minutes";
```

Reply:
I can think of a couple of example where those retries can be beneficial over longer period:

- If something is broken on EI environment on weekend, people might not jump in to repair it for a couple of days.
- While retries are happening, there is a persistent stream of error messages, they will continuously remind SREs that the problem is still happening. If one of the processing thread dies, service will just go quite. Then it will be hard to tell why exactly it has died and when.

Reply:
Although, it's worth checking how this retryer logic works. If it tries to accumulate all exceptions that ever occurred, then it will eventually run out of memory with infinite reties and crash the process.

Reply:
good suggestion...  seems there's no construction of a retry chain; rather the last attempt/exception overwrites the prior one, so no memory growth:
https://github.com/rholder/guava-retrying/blob/master/src/main/java/com/github/rholder/retry/Retryer.java#L155

Reply:
as for how long/whether to terminate... I agree on possibly a >24 hour lag before getting attention (although, again, the timeout could be configured higher).  and while a steady stream of error messages would hopefully get attention, even more important might be to alarm on the topic offset lag.  that's where I've figured the monitoring should sit.  the question I considered was whether after 24 hours the system would be likely to resolve on its own, through retries, and I judged that unlikely.  overall, ITO getting attention, I'd recommend setting the alarm on the kafka topic offset lag to catch when (at average rate) we might fall behind 30 mins of (per-partition) msgs.

what do you think?

overall, I'm not against setting the default value higher.  but because I'd like to make the maximum (timeout) configurable, it's easiest to presume it's always finite.  on your preference, we could choose a higher value, like 7 * 24 hours, whatever we feel is just too long to continue retrying (certainly before the kafka retention would expire!)